pyanaconda/storage/osinstall.py

```diff
@@ -421,13 +421,22 @@ def _get_container_name_template(self, prefix=None):
     def turn_on_swap(self):
         self.fsset.turn_on_swap(root_path=util.getSysroot())
 
-    def mount_filesystems(self, read_only=None, skip_root=False):
-        self.fsset.mount_filesystems(root_path=util.getSysroot(),
-                                     read_only=read_only, skip_root=skip_root)
+    def mount_filesystems(self):
+        root_path = util.getTargetPhysicalRoot()
+
+        # Mount the root and the filesystems.
+        self.fsset.mount_filesystems(root_path=root_path)
+
+        # Set up the sysroot.
+        util.setSysroot(root_path)
```

Comment:
What I found myself wondering about this whole patch is: "what code sets things up so that physical_root = system_root in the dnf (i.e. not ostree) path"?

And it looks like it's this bit of code - correct?

Then what the rpmostreepayload.py is relying on is calling `setSysroot()` changing that.

That sound correct?

Reply:
Yes, Anaconda calls `mount_filesystems` for all types of installation, so we always set up the system root to point at the physical root. Then the RPM OSTree payload changes it.

I was considering to skip the first step for RPM OSTree, but then the system root wouldn't be set up until the payload installation. That diverges from the current solution and might require more fixes.

Reply:
Could you please make this more obvious. It would be great to have a comment which will explain why is this required.